### DISCUSSION TITLE
Route group ->where argument type differs

### DISCUSSION THREAD

**Discussion by dennisprudlo** [laravel/framework#33704] [General] [answered]
Hello,

When using `->where(...)` to describe a route parameter in the routes file I usually use the first argument as the variable name and the second as the regular expression, for instance:

`Route::get('users/{userId}',	'ABC@xyz')->where('userId', '[0-9]+');`

When using a prefixed route group with a route parameter in it the where function only allows to pass an array:
```
Route::prefix('users/{userId}')->where(['userId' => '[0-9]+'])->group(function () {
    Route::get('/', 'ABC@xyz');
});
```

Is there any special reason that
```
Route::prefix('users/{userId}')->where('userId', '[0-9]+')->group(function () {
    Route::get('/', 'ABC@xyz');
});
```
throws `array_merge(): Expected parameter 2 to be an array, string given`?

**Comment by dkulyk** [comment, marked as answer]
`where` in this case is attribute of group and allows only one parameter.
`where(array  $where): RouteRegistrar`

```php
Route::group([
    'prefix' => 'users/{userId}',
    'where' => ['userId' => '[0-9]+']
], function () { ... })
```

**Reply by dennisprudlo** [reply]
Ah I see, so its a entirely different function. Thanks for the fast answer!
